packages/db/src/query/user.ts

```diff
@@ -126,8 +133,60 @@ type UpdateUserData = {
     warmwasser: (typeof userData.warmwasser.enumValues)[number];
     household: number;
     basispreis: number;
+    timestamp: Date;
 };
 
 export async function updateUserData(data: UpdateUserData, id: number) {
     return await db.update(userData).set(data).where(eq(userData.userId, id));
 }
+
+export async function copyOldUserDataToHistoryUserData(userId: number) {
```

Comment:
Aus meiner Sicht bräuchte man die Methode nicht zu exportieren. Es würde ja reichen, immer dann, wenn eine Update-Methode für UserData in dieser Datei ausgeführt wird, einmal vorher diese Methode auszuführen. Im Prinzip ist das ja auch ein Implementierungsdetail, über das sich die Aufrufer keine Gedanken machen sollten.
Ein weiterer Vorteil ist, dass man das Kopieren in die History-Tabelle in derselben Transaktion wie das eigentliche Update ausführen kann. Mit der aktuellen Implementierung wäre es nämlich möglich, dass der Backup in die History-Tabelle durchläuft, das Update aber nicht. Dann hat man das Backup umsonst gemacht. Prinzipiell ist das zwar nicht schlimm, aber das könnte später die Auswertung verfälschen.